### ISSUE TITLE
IllegalArgumentExpection after upgrade

### ISSUE THREAD

**Issue by GC-Xi** [snowplow/snowplow-android-tracker#418]
**Describe the bug**
In 1.3.5, you guys forced `builder.id` to be an UUID.
```
    protected ScreenView(ScreenView.Builder<?> builder) {
        super(builder);
        if (builder.id != null) {
            Preconditions.checkArgument(Util.isUUIDString(builder.id));
            this.id = builder.id;
        } else {
            this.id = Util.getUUIDString();
        }
        ...
    }
```
Is this really necessary? We have been using normal strings as id for the past few years and it will mess up our data if we change the name of the id now.

**Comment by AlexBenny** [comment]
Hi @GC-Xi

The version 1.3.5 is actually a patch we introduced to fix an older issue related to the screenID. Since the version 1.1.0 the ScreenView events rely on this schema (https://github.com/snowplow/iglu-central/blob/428b7c3da60a6def2905e092ed1b973a5fd1d21c/schemas/com.snowplowanalytics.mobile/screen_view/jsonschema/1-0-0#L22) where the screenID has to be a UUID string.
In the version 1.3.5 of the tracker we checked upfront that the id is a correct UUID in order to avoid to track those ScreenView events as _bad events_ down in the pipeline.

I guess you are upgrading from a version older than 1.1.0. There the schema used for ScreenView was different: https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/screen_view/jsonschema/1-0-0

We are not planning to change this in the future, so also the next 2.0 version is going to require a UUID string for the ScreenView ID.
